src/components/DynamicComponent/index.js

```diff
@@ -36,31 +36,38 @@ class DynamicComponent extends Component {
     this.state = {}
   }
   componentDidMount() {
-    this.props
-      .require(this.props.src)
-      .then(module => {
-        this.setState({
-          LoadedComponent: module.hasOwnProperty('default')
-            ? module['default']
-            : module
+    if (this.props.src) {
+      this.props
+        .require(this.props.src)
+        .then(module => {
+          this.setState({
+            LoadedComponent: module.hasOwnProperty('default')
+              ? module['default']
+              : module
+          })
         })
-      })
-      .catch(error => {
-        this.setState({ error })
-      })
+        .catch(error => {
+          this.setState({ error })
+        })
+    }
   }
   render() {
     const { LoadedComponent } = this.state
-
+    const IdentifierComponent = this.props.identifiers[this.props.identifier]
```

Comment:
One more thing, let's ensure they are mutually exclusive:
```suggestion
    const IdentifierComponent = !this.props.src && this.props.identifiers[this.props.identifier]
```